### PR TITLE
Jira #788:  BLE Peripheral cannot discover Central attributes, Git #382.

### PR DESCRIPTION
**Pull request by SidLeung**
1. Add test sketches discoveratperipheral.ino and profileatcentral.ino
2. Modify BLEDeviceManager.cpp to add central to profile buffer in peripheral
    role.

### CODE REVIEW

**Comment by SidLeung** [comment]
@bigdinotech @eriknyquist , please review the code change.  @noelpaz @russmcinnis, this fix has been unit tested, please performance verification testing.  Please note the additional of test sketches to exercise the fix.

File: libraries/CurieBLE/examples/central/CentralDouble/CentralDouble.ino [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/central/CentralDouble/CentralDouble.ino]: Sketch code looks good, it's pretty readable. However I have no idea what this sketch is doing, there are not really any comments to explain it.

Can you please follow the style of our other example sketches (here is a good example: https://github.com/01org/corelibs-arduino101/blob/master/libraries/CurieIMU/examples/Accelerometer/Accelerometer.ino)

Specifically, 1) put the main part of the license at the bottom of the file, with a short 1-line reference at the top. Just copy the style of existing sketches, here. And 2), add a short comment at the top, explaining what the sketch does. Again, just copy the style of existing examples.

**Comment by SidLeung** [comment]
@eriknyquist , please check out the changes per requested.  Thanks.

**Comment by SidLeung** [comment]
@eriknyquist , please check out the additional changes to the sketches under the Central and Peripheral.  They should be in the Arduino style with additional header block comments.

**Comment by eriknyquist** [comment]
What about discoveratperipheral.ino and profileatcentral.ino?

**Comment by SidLeung** [comment]
@eriknyquist , you are correct...  The sketches in the test directory are from various sources.  And, according to @noelpaz , Arduino will not release them as their official CoreLibs release.  @calvinatintel , @yashaswini-hanji , will not allow me to put anything into the release without proper licensing.  Thus, any sketches without licensing, they will be removed.

**Comment by eriknyquist** [comment]
These two sketches do not match the style of the others; there is no description (so I do not know what they are supposed to do), and the full license is at the top, unlike the others. Please fix these things.

**Comment by eriknyquist** [comment]
profileatcentral.ino doesn't even have a license at all. Just make all the example sketches look the same, with a description and license. That's all.

**Comment by SidLeung** [comment]
@eriknyquist , scanned the example sketches under the CurieBLE/example directory.  They have proper licensing and format.  For those sketches that do not have, and I can't locate, the licensing header, they are removed.  Please review.

**Comment by SidLeung** [comment]
@eriknyquist , fixed the last sketch licensing block issue.  Please review.

File: libraries/CurieBLE/examples/peripheral/led/led.ino [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
+ * Copyright (c) 2016 Intel Corporation.  All rights reserved.
+ * See the bottom of this file for the license terms.
+ */
 
-  You should have received a copy of the GNU Lesser General Public
-  License along with this library; if not, write to the Free Software
-  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
-*/
+/*
+ * Sketch: BatteryMonitor_Central.ino
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/peripheral/led/led.ino]: oops-- wrong name here (BatteryMonitor_Central.ino)

File: libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
+ * Copyright (c) 2016 Intel Corporation.  All rights reserved.
+ * See the bottom of this file for the license terms.
+ */
 
-  You should have received a copy of the GNU Lesser General Public
-  License along with this library; if not, write to the Free Software
-  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
-*/
+/*
+ * Sketch: BatteryMonitor_Central.ino
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino]: Wrong name again? BatteryMonitor_Central.ino?

File: libraries/CurieBLE/examples/central/peripheral_explorer/peripheral_explorer.ino [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-  but WITHOUT ANY WARRANTY; without even the implied warranty of
-  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-  Lesser General Public License for more details.
-
-  You should have received a copy of the GNU Lesser General Public
-  License along with this library; if not, write to the Free Software
-  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
-*/
+/*
+ * Sketch: BatteryMonitor_Central.ino
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/central/peripheral_explorer/peripheral_explorer.ino]: Our old friend BatteryMonitor_Central.ino popping up again... 😄

File: libraries/CurieBLE/examples/central/CentralDouble/CentralDouble.ino [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        delay(1000);
+    }
+    Serial.print("Disconnected");
+    Serial.println(peripheral.address());
+}
+
+
+
+/*
+  Arduino BLE Peripheral Double read/write test example
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/central/CentralDouble/CentralDouble.ino]: Wrong example sketch name here (I think?)

File: libraries/CurieBLE/examples/test/profileatcentral/profileatcentral.ino [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+			Serial.println("Failed to connect!");
+		}
+        delay (4000);
+        BLE.scanForName("LED");
+    }
+}
+
+
+/*
+  Arduino BLE Peripheral LED example
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/test/profileatcentral/profileatcentral.ino]: The name here doesn't seem to match description of example sketch at the top....

File: libraries/CurieBLE/examples/test/discoveratperipheral/discoveratperipheral.ino [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        ledCharacteristic.write(&ledstate, sizeof(ledstate));
+        delay(5000);
+    }
+    Serial.print("Disconnected");
+    Serial.println(central.address());
+}
+
+
+/*
+  Arduino BLE Peripheral LED example
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/test/discoveratperipheral/discoveratperipheral.ino]: Wrong-looking name again

**Comment by SidLeung** [comment]
@eriknyquist , licensing correction made.

**Comment by eriknyquist** [comment]
@SidLeung please see my comments about sketches wrongly named as "BatteryMonitor_Central.ino"

**Comment by SidLeung** [comment]
@eriknyquist , corrected sketch reference name in comment block.

File: libraries/CurieBLE/examples/peripheral/BatteryMonitor_Notification/BatteryMonitor_Notification.ino [hunk 8]
```diff
@@ -3,18 +3,22 @@
...
 /*
-  This sketch can work with BatteryMonitor_Central.
-
-  You can also use an android or IOS app that supports notifications.
-  This sketch example partially implements the standard Bluetooth Low-Energy Battery service
-  and connection interval paramater update.
-  For more information: https://developer.bluetooth.org/gatt/services/Pages/ServicesHome.aspx
-*/
-
+ * Sketch: led.ino
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/peripheral/BatteryMonitor_Notification/BatteryMonitor_Notification.ino]: Wrong sketch name?

**Comment by SidLeung** [comment]
@eriknyquist , sketch name corrected.  Thanks.

**Comment by SidLeung** [comment]
@yashaswini-hanji , please cherry-pick this PR to merge with main trunk.  Thanks.

**Comment by SidLeung** [comment]
@yashaswini-hanji , have cherry-picked this PR on top latest 01org/master.  Would you please merge?  Thanks.

**Comment by yashaswini-hanji** [comment]
Changes merged to base branch.

**Pull request merged**